### PR TITLE
[REF] minor refactor towards removing complexity.

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Minor code legibility improvement.

Before
----------------------------------------
Less readable

After
----------------------------------------
More readable.

Technical Details
----------------------------------------
This change actually makes the foreach & if following it redundant. I left them out for legibility
but basically all the wrangling is on the 'other' contact. We do just one thing with the 'main'
contact so the whole foreach & if is silly. Will follow up with removal of those

Comments
----------------------------------------

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
unrelated fail

**Comment by eileenmcnaughton** [comment]
test this please

File: CRM/Dedupe/Merger.php [hunk 1]
```diff
@@ -1091,10 +1091,12 @@ public static function getRowsElementsAndInfo($mainId, $otherId, $checkPermissio
         // CRM-15681 don't display sub-types in UI
         continue;
       }
-      foreach (['main' => $main, 'other' => $other] as $moniker => $contact) {
-        list($label, $value) = self::getFieldValueAndLabel($field, $contact);
-        $rows["move_$field"][$moniker] = $label;
+      $rows["move_$field"]['main'] = self::getFieldValueAndLabel($field, $main)['label'];
+      $rows["move_$field"]['other'] = self::getFieldValueAndLabel($field, $other)['label'];
+
+      foreach (['other' => $other] as $moniker => $contact) {
```
Comment (colemanw) [inline CRM/Dedupe/Merger.php]: Err, this is a foreach with only one item in the array, followed by a conditional checking to see if we are looping on the one and only item. Surely both of those can be eliminated now.

Comment (eileenmcnaughton) [reply CRM/Dedupe/Merger.php]: @colemanw yes - I did the version with it removed here https://github.com/civicrm/civicrm-core/pull/15594 but it's hard to read that one hence I created an interim PR - see PR template

**Pull request merged**